baseclass/src/main/java/com/google/apigee/calloutbaseclass/CalloutsBase.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * <p>Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * <p>https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * <p>Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.apigee.calloutbaseclass;
+
+import com.apigee.flow.message.MessageContext;
+
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/*
+ * Base Class for Apigee Java Callouts containing commonly used methods across callouts.
+ */
+public abstract class CalloutsBase {
+
+  private static final String ERROR_FLOW_VARIABLE = "callout_error";
+  private static final String EXCEPTION_FLOW_VARIABLE = "callout_exception";
+  private static final String EXCEPTION_STACKTRACE_FLOW_VARIABLE = "callout_exception_stacktrace";
+  private static final String LOG_FLOW_VARIABLE = "callout_log";
+
+  // Matches common error strings (e.g. matches "example: error")
+  private static final String commonErrorPatternString = "^(.+?)[:;] (.+)$";
```

Comment:
See my comment below about the regular expression. I think we can do something similar here.